src/app/admin/export-data/export-data.component.ts

```diff
@@ -0,0 +1,47 @@
+import {Component, Input} from '@angular/core';
+import {BackupService} from '../services/backup.service';
+
+@Component({
+  selector: 'app-export-data',
+  templateUrl: './export-data.component.html',
+  styleUrls: ['./export-data.component.scss']
+})
+export class ExportDataComponent {
+
+  @Input() data: any = [];
+  // What kind of data should be export? Currently implemented are 'json', 'csv'
+  @Input() format: string = 'csv';
+  @Input() filename: string = 'exportedData';
+
+  constructor(private backupService: BackupService) { }
+
+  exportData() {
+    const blobData = this.getFormattedBlobData();
+    const link = this.createDownloadLink(blobData);
+    link.click();
+  }
+
+  private createDownloadLink(blobData): HTMLAnchorElement {
+    const link = document.createElement('a');
+    link.setAttribute('style', 'display:none;');
+    document.body.appendChild(link);
+    link.href = window.URL.createObjectURL(blobData);
```

Comment:
We should make sure to properly destroy the object url, as it's lifecycle is bound to `document`. This might be a memory leak, if the user exports lots of things.

https://developer.mozilla.org/en-US/docs/Web/API/URL/revokeObjectURL
https://angular.io/api/core/OnDestroy

Reply:
I implemented something, but I am not quite sure how to test if it actually works.